tests/test_other.py

```diff
@@ -8813,6 +8813,65 @@ def test_main_module_without_main(self):
     run_process([PYTHON, EMCC, 'src.c', '--pre-js', 'pre.js', '-s', 'MAIN_MODULE=2'])
     self.assertContained('bar', run_js('a.out.js'))
 
+  def test_dynamicLibraries_i64(self):
+    create_test_file('pre.js', r'''
+if (typeof(Module) === "undefined") Module = {};
+Module.dynamicLibraries = ['side.wasm'];
+''')
```

Comment:
Can you just do `self.set_setting('RUNTIME_LINKED_LIBS', ['wide.wasm'])` instead of this?

Reply:
That only works in `core`, I believe? (We call emcc directly here.)

Reply:
Ah, well I guess that begs the question why don't we have common patter for dylink testing in "other"?

For now, why not just add a quick, minimal dylink-style test to 'core'?

If you think that this doesn't belong in core, the same can probably be said for a lot of the other "dylink" tests too and we can do a followup change to migrate them out of core.   I think this test might actually belong in core because the JS interaction stuff is effected by things like optimization levels.

Basically I think there is way too much boiler plate in this test which it seems to me should just test for i64 function being exported from a side module.  But maybe I'm missing something?

Reply:
Not directly, you're right, this was killing two birds with one stone :smile: 

Note that I actually tried to write a quick, minimal test - and failed. After 15 minutes I felt it wasn't a good use of time, and used the existing testcase, again, given I had realized I wanted to write a new testcase for `Module.dynamicLibraries` anyhow which it uses.

I'm not sure why I failed, as the function pointer issue should be separable from how the library is loaded - but maybe that assumption is wrong? When we dlopen vs `Module.dynamicLibraries` do we add the side module symbols differently to main somehow?

Reply:
Hmm, I'm not suggesting you use `dlopen` which I agree is a different use case.

My understanding was/is that `RUNTIME_LINKED_LIBS` and `Module.dynamicLibraries` do exactly the same thing.   I see the former as the official way to set the later.

The tests in core that use `dylink_test` use `RUNTIME_LINKED_LIBS` and not `dlopen` IIRC so they should behave exactly like adding to `Module.dynamicLibraries`.. which is to say they do library loading *before* main is called.